clients/src/main/java/org/apache/kafka/common/MetricName.java

```diff
@@ -81,6 +81,7 @@ public MetricName(String name, String group, String description, Map<String, Str
      * @param description   A human-readable description to include in the metric
      * @param keyValue      additional key/value attributes of the metric (must come in pairs)
      */
+    @Deprecated
```

Comment:
Certainly. This is useful. I have updated the comment to provide link to respective org.apache.kafka.common.metrics.Metrics#metricName(...). The code is rebased against trunk. 

The patch passed style check. I am running testAll now. It shouldn't cause new failure since the only change is in comment.